### ISSUE TITLE
Encountered CHECK failed during run

### ISSUE THREAD

**Issue by Babel-Fish** [google/sanitizers#1002]
Today after pulling latest from our repo master branch I started encountered a weird ASAN error and I have not found much online as to how to proceed/debug. It looks like ASAN is failing to obtain the thread id. I am not sure why this would be the case. Any help would be great. Thanks in advance.

Closest match I could find looks like this line: https://github.com/llvm-mirror/compiler-rt/blob/release_33/lib/asan/asan_rtl.cc#L405

```
==3282==Parsed ASAN_OPTIONS: debug=1:verbosity=3:detect_leaks=0:fast_unwind_on_fatal=1:halt_on_error=0:log_path=/home/user/repos/project/prefix
==3282==AddressSanitizer: failed to intercept 'textdomain'
==3282==AddressSanitizer: failed to intercept 'strcmp'
==3282==AddressSanitizer: failed to intercept 'strncmp'
==3282==AddressSanitizer: failed to intercept 'strcasecmp'
==3282==AddressSanitizer: failed to intercept 'strncasecmp'
==3282==AddressSanitizer: failed to intercept 'read'
==3282==AddressSanitizer: failed to intercept 'pread'
==3282==AddressSanitizer: failed to intercept 'pread64'
==3282==AddressSanitizer: failed to intercept 'readv'
==3282==AddressSanitizer: failed to intercept 'preadv'
==3282==AddressSanitizer: failed to intercept 'preadv64'
==3282==AddressSanitizer: failed to intercept 'write'
==3282==AddressSanitizer: failed to intercept 'pwrite'
==3282==AddressSanitizer: failed to intercept 'pwrite64'
==3282==AddressSanitizer: failed to intercept 'writev'
==3282==AddressSanitizer: failed to intercept 'pwritev'
==3282==AddressSanitizer: failed to intercept 'pwritev64'
==3282==AddressSanitizer: failed to intercept 'prctl'
==3282==AddressSanitizer: failed to intercept 'localtime'
==3282==AddressSanitizer: failed to intercept 'localtime_r'
==3282==AddressSanitizer: failed to intercept 'gmtime'
==3282==AddressSanitizer: failed to intercept 'gmtime_r'
==3282==AddressSanitizer: failed to intercept 'ctime'
==3282==AddressSanitizer: failed to intercept 'ctime_r'
==3282==AddressSanitizer: failed to intercept 'asctime'
==3282==AddressSanitizer: failed to intercept 'asctime_r'
==3282==AddressSanitizer: failed to intercept 'strptime'
==3282==AddressSanitizer: failed to intercept 'scanf'
==3282==AddressSanitizer: failed to intercept 'sscanf'
==3282==AddressSanitizer: failed to intercept 'fscanf'
==3282==AddressSanitizer: failed to intercept 'vscanf'
==3282==AddressSanitizer: failed to intercept 'vsscanf'
==3282==AddressSanitizer: failed to intercept 'vfscanf'
==3282==AddressSanitizer: failed to intercept '__isoc99_scanf'
==3282==AddressSanitizer: failed to intercept '__isoc99_sscanf'
==3282==AddressSanitizer: failed to intercept '__isoc99_fscanf'
==3282==AddressSanitizer: failed to intercept '__isoc99_vscanf'
==3282==AddressSanitizer: failed to intercept '__isoc99_vsscanf'
==3282==AddressSanitizer: failed to intercept '__isoc99_vfscanf'
==3282==AddressSanitizer: failed to intercept 'frexp'
==3282==AddressSanitizer: failed to intercept 'frexpf'
==3282==AddressSanitizer: failed to intercept 'frexpl'
==3282==AddressSanitizer: failed to intercept 'getpwnam'
==3282==AddressSanitizer: failed to intercept 'getpwuid'
==3282==AddressSanitizer: failed to intercept 'getgrnam'
==3282==AddressSanitizer: failed to intercept 'getgrgid'
==3282==AddressSanitizer: failed to intercept 'getpwnam_r'
==3282==AddressSanitizer: failed to intercept 'getpwuid_r'
==3282==AddressSanitizer: failed to intercept 'getgrnam_r'
==3282==AddressSanitizer: failed to intercept 'getgrgid_r'
==3282==AddressSanitizer: failed to intercept 'clock_getres'
==3282==AddressSanitizer: failed to intercept 'clock_gettime'
==3282==AddressSanitizer: failed to intercept 'clock_settime'
==3282==AddressSanitizer: failed to intercept 'getitimer'
==3282==AddressSanitizer: failed to intercept 'setitimer'
==3282==AddressSanitizer: failed to intercept 'time'
==3282==AddressSanitizer: failed to intercept 'glob'
==3282==AddressSanitizer: failed to intercept 'glob64'
==3282==AddressSanitizer: failed to intercept 'wait'
==3282==AddressSanitizer: failed to intercept 'waitid'
==3282==AddressSanitizer: failed to intercept 'waitpid'
==3282==AddressSanitizer: failed to intercept 'wait3'
==3282==AddressSanitizer: failed to intercept 'wait4'
==3282==AddressSanitizer: failed to intercept 'inet_ntop'
==3282==AddressSanitizer: failed to intercept 'inet_pton'
==3282==AddressSanitizer: failed to intercept 'pthread_getschedparam'
==3282==AddressSanitizer: failed to intercept 'getaddrinfo'
==3282==AddressSanitizer: failed to intercept 'getnameinfo'
==3282==AddressSanitizer: failed to intercept 'getsockname'
==3282==AddressSanitizer: failed to intercept 'gethostent'
==3282==AddressSanitizer: failed to intercept 'gethostbyaddr'
==3282==AddressSanitizer: failed to intercept 'gethostbyname'
==3282==AddressSanitizer: failed to intercept 'gethostbyname2'
==3282==AddressSanitizer: failed to intercept 'gethostent_r'
==3282==AddressSanitizer: failed to intercept 'gethostbyaddr_r'
==3282==AddressSanitizer: failed to intercept 'gethostbyname_r'
==3282==AddressSanitizer: failed to intercept 'gethostbyname2_r'
==3282==AddressSanitizer: failed to intercept 'getsockopt'
==3282==AddressSanitizer: failed to intercept 'accept'
==3282==AddressSanitizer: failed to intercept 'accept4'
==3282==AddressSanitizer: failed to intercept 'modf'
==3282==AddressSanitizer: failed to intercept 'modff'
==3282==AddressSanitizer: failed to intercept 'modfl'
==3282==AddressSanitizer: failed to intercept 'recvmsg'
==3282==AddressSanitizer: failed to intercept 'getpeername'
==3282==AddressSanitizer: failed to intercept 'ioctl'
==3282==AddressSanitizer: failed to intercept 'inet_aton'
==3282==AddressSanitizer: failed to intercept 'sysinfo'
==3282==AddressSanitizer: failed to intercept 'readdir'
==3282==AddressSanitizer: failed to intercept 'readdir_r'
==3282==AddressSanitizer: failed to intercept 'readdir64'
==3282==AddressSanitizer: failed to intercept 'readdir64_r'
==3282==AddressSanitizer: failed to intercept 'ptrace'
==3282==AddressSanitizer: failed to intercept 'setlocale'
==3282==AddressSanitizer: failed to intercept 'getcwd'
==3282==AddressSanitizer: failed to intercept 'get_current_dir_name'
==3282==AddressSanitizer: failed to intercept 'strtoimax'
==3282==AddressSanitizer: failed to intercept 'strtoumax'
==3282==AddressSanitizer: failed to intercept 'mbstowcs'
==3282==AddressSanitizer: failed to intercept 'mbsrtowcs'
==3282==AddressSanitizer: failed to intercept 'mbsnrtowcs'
==3282==AddressSanitizer: failed to intercept 'wcstombs'
==3282==AddressSanitizer: failed to intercept 'wcsrtombs'
==3282==AddressSanitizer: failed to intercept 'wcsnrtombs'
==3282==AddressSanitizer: failed to intercept 'tcgetattr'
==3282==AddressSanitizer: failed to intercept 'realpath'
==3282==AddressSanitizer: failed to intercept 'canonicalize_file_name'
==3282==AddressSanitizer: failed to intercept 'confstr'
==3282==AddressSanitizer: failed to intercept 'sched_getaffinity'
==3282==AddressSanitizer: failed to intercept 'strerror'
==3282==AddressSanitizer: failed to intercept 'strerror_r'
==3282==AddressSanitizer: failed to intercept '__xpg_strerror_r'
==3282==AddressSanitizer: failed to intercept 'scandir'
==3282==AddressSanitizer: failed to intercept 'scandir64'
==3282==AddressSanitizer: failed to intercept 'getgroups'
==3282==AddressSanitizer: failed to intercept 'poll'
==3282==AddressSanitizer: failed to intercept 'ppoll'
==3282==AddressSanitizer: failed to intercept 'wordexp'
==3282==AddressSanitizer: failed to intercept 'sigwait'
==3282==AddressSanitizer: failed to intercept 'sigwaitinfo'
==3282==AddressSanitizer: failed to intercept 'sigtimedwait'
==3282==AddressSanitizer: failed to intercept 'sigemptyset'
==3282==AddressSanitizer: failed to intercept 'sigfillset'
==3282==AddressSanitizer: failed to intercept 'sigpending'
==3282==AddressSanitizer: failed to intercept 'sigprocmask'
==3282==AddressSanitizer: failed to intercept 'backtrace'
==3282==AddressSanitizer: failed to intercept 'backtrace_symbols'
==3282==AddressSanitizer: failed to intercept '_exit'
==3282==AddressSanitizer: failed to intercept 'pthread_mutex_lock'
==3282==AddressSanitizer: failed to intercept 'pthread_mutex_unlock'
==3282==AddressSanitizer: failed to intercept 'getmntent'
==3282==AddressSanitizer: failed to intercept 'getmntent_r'
==3282==AddressSanitizer: failed to intercept 'statfs'
==3282==AddressSanitizer: failed to intercept 'fstatfs'
==3282==AddressSanitizer: failed to intercept 'statfs64'
==3282==AddressSanitizer: failed to intercept 'fstatfs64'
==3282==AddressSanitizer: failed to intercept 'statvfs'
==3282==AddressSanitizer: failed to intercept 'fstatvfs'
==3282==AddressSanitizer: failed to intercept 'statvfs64'
==3282==AddressSanitizer: failed to intercept 'fstatvfs64'
==3282==AddressSanitizer: failed to intercept 'initgroups'
==3282==AddressSanitizer: failed to intercept 'ether_ntoa'
==3282==AddressSanitizer: failed to intercept 'ether_aton'
==3282==AddressSanitizer: failed to intercept 'ether_ntohost'
==3282==AddressSanitizer: failed to intercept 'ether_hostton'
==3282==AddressSanitizer: failed to intercept 'ether_line'
==3282==AddressSanitizer: failed to intercept 'ether_ntoa_r'
==3282==AddressSanitizer: failed to intercept 'ether_aton_r'
==3282==AddressSanitizer: failed to intercept 'shmctl'
==3282==AddressSanitizer: failed to intercept 'random_r'
==3282==AddressSanitizer: failed to intercept 'pthread_attr_getdetachstate'
==3282==AddressSanitizer: failed to intercept 'pthread_attr_getguardsize'
==3282==AddressSanitizer: failed to intercept 'pthread_attr_getschedparam'
==3282==AddressSanitizer: failed to intercept 'pthread_attr_getschedpolicy'
==3282==AddressSanitizer: failed to intercept 'pthread_attr_getscope'
==3282==AddressSanitizer: failed to intercept 'pthread_attr_getstacksize'
==3282==AddressSanitizer: failed to intercept 'pthread_attr_getstack'
==3282==AddressSanitizer: failed to intercept 'pthread_attr_getinheritsched'
==3282==AddressSanitizer: failed to intercept 'pthread_attr_getaffinity_np'
==3282==AddressSanitizer: failed to intercept 'tmpnam'
==3282==AddressSanitizer: failed to intercept 'tmpnam_r'
==3282==AddressSanitizer: failed to intercept 'tempnam'
==3282==AddressSanitizer: failed to intercept 'pthread_setname_np'
==3282==AddressSanitizer: failed to intercept 'sincos'
==3282==AddressSanitizer: failed to intercept 'sincosf'
==3282==AddressSanitizer: failed to intercept 'sincosl'
==3282==AddressSanitizer: failed to intercept 'remquo'
==3282==AddressSanitizer: failed to intercept 'remquof'
==3282==AddressSanitizer: failed to intercept 'remquol'
==3282==AddressSanitizer: failed to intercept 'lgamma'
==3282==AddressSanitizer: failed to intercept 'lgammaf'
==3282==AddressSanitizer: failed to intercept 'lgammal'
==3282==AddressSanitizer: failed to intercept 'lgamma_r'
==3282==AddressSanitizer: failed to intercept 'lgammaf_r'
==3282==AddressSanitizer: failed to intercept 'lgammal_r'
==3282==AddressSanitizer: failed to intercept 'drand48_r'
==3282==AddressSanitizer: failed to intercept 'lrand48_r'
==3282==AddressSanitizer: failed to intercept 'getline'
==3282==AddressSanitizer: failed to intercept 'getdelim'
==3282==AddressSanitizer: failed to intercept 'iconv'
==3282==AddressSanitizer: failed to intercept 'times'
==3282==AddressSanitizer: failed to intercept 'memcmp'
==3282==AddressSanitizer: failed to intercept 'memmove'
==3282==AddressSanitizer: failed to intercept 'memset'
==3282==AddressSanitizer: failed to intercept 'memcpy'
==3282==AddressSanitizer: failed to intercept 'strcat'
==3282==AddressSanitizer: failed to intercept 'strchr'
==3282==AddressSanitizer: failed to intercept 'strcpy'
==3282==AddressSanitizer: failed to intercept 'strlen'
==3282==AddressSanitizer: failed to intercept 'wcslen'
==3282==AddressSanitizer: failed to intercept 'strncat'
==3282==AddressSanitizer: failed to intercept 'strncpy'
==3282==AddressSanitizer: failed to intercept 'strdup'
==3282==AddressSanitizer: failed to intercept 'strnlen'
==3282==AddressSanitizer: failed to intercept 'index'
==3282==AddressSanitizer: failed to intercept 'atoi'
==3282==AddressSanitizer: failed to intercept 'atol'
==3282==AddressSanitizer: failed to intercept 'strtol'
==3282==AddressSanitizer: failed to intercept 'atoll'
==3282==AddressSanitizer: failed to intercept 'strtoll'
==3282==AddressSanitizer: failed to intercept 'mlock'
==3282==AddressSanitizer: failed to intercept 'munlock'
==3282==AddressSanitizer: failed to intercept 'mlockall'
==3282==AddressSanitizer: failed to intercept 'munlockall'
==3282==AddressSanitizer: failed to intercept 'longjmp'
==3282==AddressSanitizer: failed to intercept 'sigaction'
==3282==AddressSanitizer: failed to intercept 'signal'
==3282==AddressSanitizer: failed to intercept 'swapcontext'
==3282==AddressSanitizer: failed to intercept '_longjmp'
==3282==AddressSanitizer: failed to intercept 'siglongjmp'
==3282==AddressSanitizer: failed to intercept 'pthread_create'
==3282==AddressSanitizer: failed to intercept '__cxa_atexit'
==3282==AddressSanitizer: libc interceptors initialized
|| `[0x10007fff8000, 0x7fffffffffff]` || HighMem    ||
|| `[0x02008fff7000, 0x10007fff7fff]` || HighShadow ||
|| `[0x00008fff7000, 0x02008fff6fff]` || ShadowGap  ||
|| `[0x00007fff8000, 0x00008fff6fff]` || LowShadow  ||
|| `[0x000000000000, 0x00007fff7fff]` || LowMem     ||
MemToShadow(shadow): 0x00008fff7000 0x000091ff6dff 0x004091ff6e00 0x02008fff6fff
red_zone=16
quarantine_size=256M
malloc_context_size=30
SHADOW_SCALE: 3
SHADOW_GRANULARITY: 8
SHADOW_OFFSET: 7fff8000
==3282==Installed the sigaction for signal 11
==3282==SetCurrentThread: 0x7f4b9972a000 for thread 0x7f4b8ce5e700
==3282==T0: stack [0x7f4b8c65f000,0x7f4b8ce5f000) size 0x800000; local=0x7f4b8ce550fc
==3282==AddressSanitizer Init done
==3282==AddressSanitizer CHECK failed: ../../../../gcc-4.9.1/libsanitizer/asan/asan_rtl.cc:411 "((curr_thread)) != (0)" (0x0, 0x0)
    <empty stack>
```

**Comment by kcc** [comment]
We'll need a full reproducer (exact OS/compiler/linker versions, test case, command line)
to understand what't going on here.

this is very strange:
```
==3282==AddressSanitizer: failed to intercept 'pthread_create'
```
I'd expect some problem with linking.

**Comment by dvyukov** [comment]
gcc-4.9.1 is more than 4 years old, dozens of bugs were fixed since then. Please try a newer compiler.

**Comment by Babel-Fish** [comment]
@dvyukov, I have been trying to upgrade our compiler but unfortunately there is a lot of moving parts to get the entire organization up to date. It is a work in progress.

@kcc I was actually able to track down the offending code. It was another team's C++ library wrapped in python bindings. I am still trying to sort through all of my findings but my guess is that even though the C++ code was instrumented, it was failing in such a way that ASAN was unable to get information out of it. When I adjusted the python code ordering/api to invoke the underlying library differently, I was able to bypass the issue. I will try to later track down what in the C++ library is really causing the problem. Thanks again.
